### PR TITLE
feat: add error visual indicator pull image input

### LINKED ISSUE #8926: pulling/searching image: notify if image is not recognized
### Is your enhancement related to a problem? Please describe

following https://github.com/containers/podman-desktop/issues/971 if I copy/paste an image name in the field I have not the typeahead indication

### Describe the solution you'd like

I would like to have a visual indication that the image is recognized (or not recognized)
like changing the color if not found or changing the color if found

### Describe alternatives you've considered

any suggestion that could help the user to have a small indicator

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gastoner**
### What does this PR do?
Adds border that will be red if the image specified does not exist

### Screenshot / video of UI

https://github.com/user-attachments/assets/d9fd79b3-2e29-44df-9f77-367006adfcbb

Follow up of https://github.com/podman-desktop/podman-desktop/pull/9782 - needs to be rebased 

### What issues does this PR fix or reference?
Closes #8926 
2/2 PRs
### How to test this PR?
Try to pull an image

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/ui/Typeahead.spec.ts [hunk 1]
```diff
@@ -47,6 +47,11 @@ function assertItemSelected(items: HTMLElement[], r: number): void {
   }
 }
 
+beforeEach(() => {
+  vi.resetAllMocks();
+  vi.restoreAllMocks();
```
Comment (benoitf) [inline packages/renderer/src/lib/ui/Typeahead.spec.ts]: not sure about the pattern using restoreallMock in beforeEach

Comment (gastoner) [reply packages/renderer/src/lib/ui/Typeahead.spec.ts]: It needs to be rebased after https://github.com/podman-desktop/podman-desktop/pull/9782 is merged, changes in this PR are just for pullImage.svelte

Comment (gastoner) [reply packages/renderer/src/lib/ui/Typeahead.spec.ts]: Updated in first PR

File: packages/renderer/src/lib/image/PullImage.svelte [hunk 2]
```diff
@@ -219,13 +220,15 @@ async function searchLatestTag(): Promise<void> {
       image = image.slice(DOCKER_PREFIX_WITH_SLASH.length);
     }
     const tags = await window.listImageTagsInRegistry({ image });
+    isValidName = Boolean(tags);
```
Comment (axel7083) [inline packages/renderer/src/lib/image/PullImage.svelte:228]: no tests ?

Comment (gastoner) [reply packages/renderer/src/lib/image/PullImage.svelte:228]: Tests for this are in https://github.com/podman-desktop/podman-desktop/pull/9782

**Review by axel7083 — APPROVED** [review]
Tested and LGTM! Great job @gastoner :rocket:

File: packages/renderer/src/lib/image/PullImage.svelte (hunk 2 above)
Comment (benoitf) [reply packages/renderer/src/lib/image/PullImage.svelte:228]: well, there is a test of TypeAhead, but we should still have tests there to check that depending on the response from listImageTagsInRegistry, isValidName can be either false or true and then the error attribute

Comment (benoitf) [reply packages/renderer/src/lib/image/PullImage.svelte:228]: TypeAhead unit test is to check that if there is error, the colors are correctly changed

but here, unit test is to check that isValidName is correctly updated based on tags result

File: packages/renderer/src/lib/image/PullImage.spec.ts [hunk 3]
```diff
@@ -362,3 +362,47 @@ test('Expect latest tag warning is not displayed when the image has latest tag',
   const errorMesssage = screen.queryByRole('alert', { name: 'Warning Message Content' });
   expect(errorMesssage).toBeNull();
 });
+
+test('Expect isValidName to be truthy === error to be falsy', async () => {
```
Comment (axel7083) [inline packages/renderer/src/lib/image/PullImage.spec.ts]: Could we avoid `===` in test name ? 

```suggestion
test('input component should not raise an error when the input is valid', async () => {
```

File: packages/renderer/src/lib/image/PullImage.spec.ts [hunk 4]
```diff
@@ -362,3 +362,47 @@ test('Expect latest tag warning is not displayed when the image has latest tag',
...
+  await userEvent.keyboard('my-registry/image');
+
+  const cellOutsideInput = pullImage.getAllByRole('textbox');
+  const parentInput = cellOutsideInput[0].parentElement;
+  expect(parentInput).not.toHaveClass('border-b-[var(--pd-input-field-stroke-error)]');
+  expect(parentInput).not.toHaveClass('focus-within:border-[var(--pd-input-field-stroke-error)]');
+  expect(parentInput).toHaveClass('hover:border-b-[var(--pd-input-field-hover-stroke)]');
+});
+
+test('Expect isValidName to be falsy === error to be truthy', async () => {
```
Comment (axel7083) [inline packages/renderer/src/lib/image/PullImage.spec.ts]: same remark

File: packages/renderer/src/lib/image/PullImage.spec.ts [hunk 5]
```diff
@@ -362,3 +362,47 @@ test('Expect latest tag warning is not displayed when the image has latest tag',
...
+  await userEvent.keyboard('my-registry/image');
+
+  const cellOutsideInput = pullImage.getAllByRole('textbox');
+  const parentInput = cellOutsideInput[0].parentElement;
+  expect(parentInput).toHaveClass('border-b-[var(--pd-input-field-stroke-error)]');
+  expect(parentInput).toHaveClass('focus-within:border-[var(--pd-input-field-stroke-error)]');
+  expect(parentInput).not.toHaveClass('hover:border-b-[var(--pd-input-field-hover-stroke)]');
+});
+
+test('Expect isValidName to be falsy by error === error variable should be truthy', async () => {
```
Comment (axel7083) [inline packages/renderer/src/lib/image/PullImage.spec.ts]: same remark

**Review by axel7083 — APPROVED** [review]

**Comment by benoitf** [comment]
if I specify a tag that is not existing, I don't see the red box

https://github.com/user-attachments/assets/252fdba2-16d3-48b6-9d29-0659b50a83f5




Using the keys, I have the red box while it shouldn't

here is a screencast:

https://github.com/user-attachments/assets/f14b4e53-dbb2-4045-9955-7675037a0235

**Review by benoitf — CHANGES_REQUESTED** [review]
on my end, I see red when it shouldn't and I see purple when I should see red

**Comment by gastoner** [comment]
@axel7083 I've added verification for tags, can you check again please?

**Comment by benoitf** [comment]
I still have an issue
image

**Comment by gastoner** [comment]
> I still have an issue image

Should be fixed, the issue was that the value was validated only when it was typed/clicked and not just selected by arrows

**Comment by gastoner** [comment]
@benoitf, can you please check again, once you have some time?

**Review by benoitf — APPROVED** [review]
checked and everything is working as expected now 👍

**Review by benoitf — COMMENTED** [review]
non-blocking: could be interesting to add unit tests covering the issue I had using the arrow keyboard keys

**Pull request merged**